chia/cmds/wallet.py

```diff
@@ -590,10 +624,18 @@ def take_offer_cmd(
 @click.option(
     "-m", "--fee", help="The fee to use when cancelling the offer securely, in XCH", default="0", show_default=True
 )
-def cancel_offer_cmd(wallet_rpc_port: Optional[int], fingerprint: int, id: str, insecure: bool, fee: str) -> None:
+@click.option(
+    "--no-confirm",
+    help="Do not ask for confirmation before canceling the offer.",
+    is_flag=True,
+    default=False,
+)
+def cancel_offer_cmd(
+    wallet_rpc_port: Optional[int], fingerprint: int, id: str, insecure: bool, fee: str, no_confirm: bool
+) -> None:
     from .wallet_funcs import cancel_offer
 
-    asyncio.run(cancel_offer(wallet_rpc_port, fingerprint, Decimal(fee), id, insecure))
+    asyncio.run(cancel_offer(wallet_rpc_port, fingerprint, Decimal(fee), id, not insecure, no_confirm))
```

Comment:
this `not` here looks like an important change in behavior. it looks like something that shouldn't be mixed up in a PR adding unit tests, and like something that should have a clear rationale

Reply:
yeah whoever made the test seemed to forget it, the command would do the wrong thing without it.

Reply:
this isn't a test though. It looks like a bug in production code. should we fix this bug in the 2.0 branch as well?

Reply:
this change looks correct though. I just think it's buried in mostly unit test code. I think it should be pulled out and maybe fixed in 2.0 if it applies as well

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/16012